cli.spec.js

```diff
@@ -0,0 +1,79 @@
+const test = require('ava');
+const fs = require('fs');
+const mkdirp = require('mkdirp');
+const rimraf = require('rimraf');
+const { exec } = require("child_process");
+
+
+test.before(t => {
+  /* Lets create this structure of folders and files
+    /test  
+      /first
+        /fourth
+          F.js
+          F.spec.js
+        A.js
+        B.js
+        C.js
+        C.spec.js
+      /second
+        D.js
+        E.js
+      /third
+        TEST.spec.js
+  */
+  const folders = [
+    './test/first/fourth',
+    './test/second',
+    './test/third',
+  ];
+  const files = [
+    './test/first/A.js',
+    './test/first/B.js',
+    './test/first/C.js',
+    './test/first/C.spec.js',
+    './test/first/fourth/F.js',
+    './test/first/fourth/F.spec.js',
+    './test/second/D.js',
+    './test/second/E.js',
+    './test/third/TEST.spec.js',
+  ];
+  return Promise.all(
+    folders.map((folder) => mkdirp(folder).catch((err) => t.log(err)))
+  ).then(() => {
+    files.forEach((file) => fs.writeFile(file, '', () => {}));
+  });
+});
+
+test.after.always('Guaranteed cleanup', () => {
```

Comment:
This seems to be more likely a unit test scenario where each test has to be aseptically to each other, In this case the propourse of this test is meant to be more like a integration test using start to end state, which each step checks the mutations of the state and transform it  expecting different results, this simplify code lines. 

I realice each test is defined as serial so thats shows up this test runner is not the ideal for this situation. A simple and alike test runner is [tape](https://www.npmjs.com/package/tape), which offer the same api with less multithread logic  

https://github.com/k14v/nlink/commits/feature/Add_test_fixtures_tape 

But this is only fo integration test, if in a soon future we're gonna introduce unit testing is much better use ava to manage all with the same test runner.

Reply:
`test.after` only run when all the test in the file are ended